Fixes bugs with map view. Changes geofences initiation order.

sophiabersani opened the pull request:
https://www.pivotaltracker.com/story/show/128020649
https://www.pivotaltracker.com/story/show/127749951
https://www.pivotaltracker.com/story/show/128162141
- Fixes maps not centering on user and staying on default location.
- Changes the order of geofencining initialization to try to prevent some bugs in some android devices.

Update:
- Removes the current custom `Loading` animation to allow RN's InteractionManager to work.
- Implements the InteractionManager API to the Map View for smoother mounting.

sophiabersani committed:
Sync the map's region with the user's location

sophiabersani committed:
Setup geofencing only after starting initial service

sophiabersani committed:
Cleans state logs

dan-nyanko commented on js/views/MapPage.js:
```diff
@@ -56,11 +56,18 @@ const MapPage = React.createClass({
-        }
-      }, 300);
+      getUserLocationData((location) => {
+        setTimeout(() => {
```
Why use a setTimeout here?

sophiabersani replied:
`InteractionManager` replacement I explained to you in a previous PR.
https://facebook.github.io/react-native/docs/interactionmanager.html

sophiabersani committed:
Adds mock location settings to manifest

- Adds redundant service restart to guarantee geofences are checked on
setup

sophiabersani committed:
Implements InteractionManager to the Map view

sophiabersani committed:
Removes JS loading animation from iOS

sophiabersani committed:
Remove unused components on iOS

dan-nyanko commented on js/api/Geofencing.js:
```diff
@@ -129,6 +130,12 @@ export function setupGeofences(callback) {
+
+        // Restarts the service on Android devices.
+        // This redundancy is a patch to force geofences to be rechecked in some devices.
+        if (Platform.OS === 'android') {
```
@gbersani What devices require this redundant restart?

sophiabersani replied:
Mine.

Although it may be related to the bug you mentioned on the comment just below.
I will remove this on the next update if my geofences resume working as intended.

dan-nyanko commented:
@gbersani - Triggers are not active upon initial acceptance of the survey and the user has to restart the application.

Steps:
1. Delete GQ app from emulator
2. Perform fresh install of this branch `bugfix-map`
3. Alter `tool-populate/data/DemoGeofenceData.js` with location data
4. Run `node tool-populate/index.js -g`
5. Run `node tool-populate/index.js -p`
6. Build GQ `react-native run-android`
7. Accept the survey created in step 3
8. Click `geofence` tab
9. Run Lockito with mock location route

Note: the geofences do not trigger.
1. Terminate the GQ app on the emulator
2. Launch GQ and repeat steps 8 and 9.

Note: the geofence triggers work.

sophiabersani commented:
These are already great news, thank you Dan.
Expect an update shortly to tackle this other issue.

sophiabersani committed:
Fixes background services not initializing.

- Fixes an issue with the background service not starting if the user
openned the app while not logged in.
- Fixes geolocation-only mode not activating on Android.

sophiabersani committed:
lint

sophiabersani committed:
Removes stop() statements when switching modes

dan-nyanko commented:
@gbersani Unfortunately the last 3 commits does not fix the initial loading problem and instead causes the geofences to not trigger at all.  

Logcat reports a lot of activity switching between modes which I believe is the cause of these inconsistencies.  I suggest removing the extra logic/calls that stop/start the location services and just stick with one mode until this is working properly.

```
8-12 12:02:13.754 19554-19647/? I/ReactNativeJS: Configuring Geolocation: Energy-Saving Mode
08-12 12:02:23.594 20416-20544/? I/ReactNativeJS: Running application "GoodQuestion" with appParams: {"initialProps":{},"rootTag":1}. __DEV__ === true, development-level warning are ON, performance optimizations are OFF
08-12 12:02:23.804 20416-20544/? I/ReactNativeJS: { currentUser: { _objCount: 0, className: '_User', id: '<removed>' },
08-12 12:02:24.674 20416-20544/? I/ReactNativeJS: Configuring Geolocation: High Accuracy Mode
08-12 12:02:25.834 20416-20544/? I/ReactNativeJS: Geolocation tracking started.
08-12 12:02:25.864 20416-20544/? I/ReactNativeJS: Cleared current geofencing settings.
08-12 12:02:25.864 20416-20544/? I/ReactNativeJS: Adding 3 geofences...
08-12 12:02:25.894 20416-20544/? I/ReactNativeJS: Successfully added geofences.
08-12 12:02:34.634 20416-20544/? I/ReactNativeJS: Configuring Geolocation: Energy-Saving Mode
08-12 12:02:47.614 20416-20544/? I/ReactNativeJS: Configuring Geolocation: High Accuracy Mode
08-12 12:03:15.234 20416-20544/? I/ReactNativeJS: Configuring Geolocation: Energy-Saving Mode
08-12 12:05:30.073 3545-3570/? I/ReactNativeJS: Running application "GoodQuestion" with appParams: {"initialProps":{},"rootTag":1}. __DEV__ === true, development-level warning are ON, performance optimizations are OFF
08-12 12:05:30.295 3545-3570/? W/ReactNativeJS: Warning: Failed prop type: Invalid prop `style` supplied to `Button`.
                                                    in Button (created by LoginPage)
                                                    in LoginPage (created by Navigator)
                                                    in RCTView (created by View)
                                                    in View (created by Navigator)
                                                    in RCTView (created by View)
                                                    in View (created by Navigator)
                                                    in RCTView (created by View)
                                                    in View (created by Navigator)
                                                    in RCTView (created by View)
                                                    in View (created by Navigator)
                                                    in Navigator (created by SharedNavigator)
                                                    in SharedNavigator (created by GoodQuestion)
                                                    in GoodQuestion
                                                    in RCTView (created by View)
                                                    in View (created by AppContainer)
                                                    in RCTView (created by View)
                                                    in View (created by AppContainer)
                                                    in AppContainer
08-12 12:05:30.296 3545-3570/? W/ReactNativeJS: Warning: Failed prop type: Invalid prop `textStyle` of type `array` supplied to `Button`, expected `object`.
                                                    in Button (created by LoginPage)
                                                    in LoginPage (created by Navigator)
                                                    in RCTView (created by View)
                                                    in View (created by Navigator)
                                                    in RCTView (created by View)
                                                    in View (created by Navigator)
                                                    in RCTView (created by View)
                                                    in View (created by Navigator)
                                                    in RCTView (created by View)
                                                    in View (created by Navigator)
                                                    in Navigator (created by SharedNavigator)
                                                    in SharedNavigator (created by GoodQuestion)
                                                    in GoodQuestion
                                                    in RCTView (created by View)
                                                    in View (created by AppContainer)
                                                    in RCTView (created by View)
                                                    in View (created by AppContainer)
                                                    in AppContainer
08-12 12:05:30.468 3545-3570/? W/ReactNativeJS: '`%s` event should be registered via the Keyboard module', 'keyboardDidShow'
08-12 12:05:30.472 3545-3570/? W/ReactNativeJS: '`%s` event should be registered via the Keyboard module', 'keyboardDidHide'
08-12 12:05:51.356 3545-3570/? W/ReactNativeJS: '`%s` event should be registered via the Keyboard module', 'keyboardDidShow'
08-12 12:05:51.356 3545-3570/? W/ReactNativeJS: '`%s` event should be registered via the Keyboard module', 'keyboardDidHide'
08-12 12:05:51.360 3545-3570/? I/ReactNativeJS: Configuring Geolocation: High Accuracy Mode
08-12 12:05:51.550 3545-3570/? I/ReactNativeJS: Geolocation tracking started.
08-12 12:05:51.763 3545-3570/? I/ReactNativeJS: Cleared current geofencing settings.
08-12 12:05:51.763 3545-3570/? I/ReactNativeJS: Adding 0 geofences...
08-12 12:05:51.767 3545-3570/? I/ReactNativeJS: Successfully added geofences.
```

sophiabersani commented:
The changes between modes are normal. It keeps the app from heavily draining the battery when it goes into the background.

`08-12 12:05:51.763 3545-3570/? I/ReactNativeJS: Adding 0 geofences...`
This however should not be happening. This means the geofence triggers were not loaded correctly from Parse. (Possibly related to #154)
I'll look into this one more time and will get you an update asap.

dan-nyanko commented:
@gbersani Thank you. I really hope to get this working soon!

sophiabersani committed:
Merge remote-tracking branch 'refs/remotes/origin/master-v2' into bugfix-map

sophiabersani committed:
Abort creating geofences if no triggers are found

sophiabersani committed:
Updates logging
